File: server/datasets/controlled_benchmarks/java/multi_student_cohorts/cohort_g1/renamed_40_bst.java
class Node_4 {
    int val_4;
    Node_4 lhs_4, rhs_4;
    public Node_4(int item) {
        val_4 = item;
        lhs_4 = rhs_4 = null;
    }
}
public class Tree_4 {
    public static void main(String[] args) {
        Tree_4 bst = new Tree_4();
        bst.ins_4(50);
        bst.srch_4(50);
    }
    Node_4 rt_4;
    public void ins_4(int val_4) {
        rt_4 = in_rec_4(rt_4, val_4);
    }
    Node_4 in_rec_4(Node_4 rt_4, int val_4) {
        if (rt_4 == null) return new Node_4(val_4);
        if (val_4 < rt_4.val_4) rt_4.lhs_4 = in_rec_4(rt_4.lhs_4, val_4);
        else if (val_4 > rt_4.val_4) rt_4.rhs_4 = in_rec_4(rt_4.rhs_4, val_4);
        return rt_4;
    }
    Node_4 srch_4(int val_4) {
        return sr_rec_4(rt_4, val_4);
    }
    Node_4 sr_rec_4(Node_4 rt_4, int val_4) {
        if (rt_4 == null || rt_4.val_4 == val_4) return rt_4;
        if (rt_4.val_4 < val_4) return sr_rec_4(rt_4.rhs_4, val_4);
        return sr_rec_4(rt_4.lhs_4, val_4);
    }
}

File: server/datasets/controlled_benchmarks/python/multiple_files/multiple_g1/renamed_40_bst.py
class Node_4:
    def __init__(self, val_4):
        self.val_4 = val_4
        self.lhs_4 = None
        self.rhs_4 = None

class Tree_4:
    def __init__(self):
        self.rt_4 = None

    def ins_4(self, val_4):
        if self.rt_4 is None:
            self.rt_4 = Node_4(val_4)
        else:
            self._ins_4_rec(self.rt_4, val_4)

    def _ins_4_rec(self, nd_4, val_4):
        if val_4 < nd_4.val_4:
            if nd_4.lhs_4 is None:
                nd_4.lhs_4 = Node_4(val_4)
            else:
                self._ins_4_rec(nd_4.lhs_4, val_4)
        elif val_4 > nd_4.val_4:
            if nd_4.rhs_4 is None:
                nd_4.rhs_4 = Node_4(val_4)
            else:
                self._ins_4_rec(nd_4.rhs_4, val_4)

    def srch_4(self, val_4):
        return self._srch_4_rec(self.rt_4, val_4)

    def _srch_4_rec(self, nd_4, val_4):
        if nd_4 is None or nd_4.val_4 == val_4:
            return nd_4
        if val_4 < nd_4.val_4:
            return self._srch_4_rec(nd_4.lhs_4, val_4)
        return self._srch_4_rec(nd_4.rhs_4, val_4)
